x-pack/plugins/alerting/server/routes/_mock_handler_arguments.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandlerContext, KibanaRequest, KibanaResponseFactory } from 'kibana/server';
+import { identity } from 'lodash';
+import { httpServerMock } from '../../../../../src/core/server/mocks';
+
+export function mockHandlerArguments(
+  { alertsClient, listTypes: listTypesRes = [] }: any,
+  req: any,
+  res?: Array<MethodKeysOf<KibanaResponseFactory>>
+): [RequestHandlerContext, KibanaRequest<any, any, any, any>, KibanaResponseFactory] {
+  const listTypes = jest.fn(() => listTypesRes);
+  return [
+    ({
+      alerting: {
+        listTypes,
+        getAlertsClient() {
+          return (
+            alertsClient || {
+              get: jest.fn(),
```

Comment:
curious why we're not using the alertClient mock here.  Also, with the lack of typing the inputs and outputs of the enclosing function, feels like it could out-of-date fairly easily.  OTOH, not worth fixing just for a mock, right now, can only cause testing issues.  We can enhance it, if possible, when that happens.  May also be the case that it's not possible or very icky to provide strict typing here.

Reply:
Fixed 👍